core/internal/features_test.go

```diff
@@ -160,7 +160,7 @@ func TestIntegration_FeeBump(t *testing.T) {
 	newHeads <- models.BlockHeader{Number: cltest.BigHexInt(confirmed)} // 23460
 	eth.EventuallyAllCalled(t)
 	jr = cltest.WaitForJobRunToPendConfirmations(t, app.Store, jr)
-	cltest.WaitForTxAttemptCount(t, app.Store, 2)
+	cltest.WaitForTxAttemptCount(t, app.Store, 3)
```

Comment:
It [looks like](https://github.com/smartcontractkit/chainlink/pull/1239/files#diff-a88d9f625a34d36439d63a134b6dd2ceR158) the `eth.Context` block before this returns a confirmed transaction receipt. When an attempt has confirmed there shouldn't be any reason to create new attempts. So why is the count increasing here?

Reply:
I see `eth_sendRawTransaction` before this three times, I made a deliberate change to this code to explicitly record every actual "attempt" as a separate record, as this is more accurate and makes for easier debugging. So I believe this expectation to be correct.

Reply:
I see. I like it, an attempt is actually a record of what was sent to the chain.